talker_agent/talker.py

```diff
@@ -870,6 +870,21 @@ def unregister_fileno(self, fileno):
             pass  # closed while processing data
 
 
+def run_diagnostics():
+    try:
+        script_path = config.parser.get('diagnostics', 'script_path')
+    except NoSectionError:
+        logger.warning('diagnostics script path is missing from config file')
+        return
+
+    if os.path.isfile(script_path):
+        logger.info("Running diagnostics")
+        try:
+            subprocess.Popen(["bash", "-ce", script_path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
I am not sure the process should use Talker's `stdout` and `stderr`. Especially when we know we are about to die.